packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1445,7 +1467,7 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
   }
 
   double get _borderWeight {
-    if (decoration.isCollapsed || decoration.border == null || !decoration.enabled)
+    if (decoration.isCollapsed || decoration.border == InputBorder.none || !decoration.enabled)
```

Comment:
i don't think InputBorder.none should be special here. If I create a new InputBorder class that is otherwise identical to _NoInputBorder, it should work the same.

Reply:
You are correct. At the moment there's no other way to recognize an InputBorder that's not actually going to draw a border.  I think could use the value of InputBorder.borderSide by giving the initial value (current BorderSide.none) some meaning. I will address this in a separate PR.

Reply:
Filed: https://github.com/flutter/flutter/issues/14197